src/main/java/bookstore/service/BookServiceImpl.java

```diff
@@ -31,9 +31,20 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto getBookById(Long id) {
-        Book book = bookRepository.getBookByID(id)
-                .orElseThrow(() -> new EntityNotFoundException("Can`t find book by id " + id)
-        );
-        return bookMapper.toDto(book);
+        return bookMapper.toDto((Book) bookRepository.getBookById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can`t find book by id " + id)));
+
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto bookRequestDto) {
+        Book book = bookMapper.toBook(bookRequestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
To make logic correct, add method to BookMapper
void updateBook(CreateBookRequestDto updatedRequestDto, @MappingTarget Book bookToUpdate); 

Then here, find book by Id in DB, use this updateBook method 
 bookMapper.updateBook(updateBookRequestDto, bookToUpdate);
 
 and then convert toDto